parastell/utils.py

```diff
@@ -202,3 +207,119 @@ def smooth_matrix(matrix, steps, sigma):
         previous_matrix = smoothed_matrix
 
     return smoothed_matrix
+
+
+class DAGMCMerger:
+    """Class to facilitate renumbering of entities to merge DAGMC models.
+
+    Arguments:
+        mb (PyMOAB Core): PyMOAB core with a DAGMC model loaded. Optional.
+    """
+
+    def __init__(self, mb=None):
+        self.mb = mb
+        if mb is None:
+            self.mb = core.Core()
+        self.geom_dim_tag = self.mb.tag_get_handle(
+            "GEOM_DIMENSION", 1, types.MB_TYPE_INTEGER, types.MB_TAG_DENSE
+        )
+        self.global_id_tag = self.mb.tag_get_handle(
+            "GLOBAL_ID", 1, types.MB_TYPE_INTEGER, types.MB_TAG_DENSE
+        )
+
+    def load_file(self, filename):
+        """Load DAGMC model from file.
+
+        Arguments:
+            filename (str): Path to DAGMC model to be loaded.
+        """
+        self.mb.load_file(filename)
+
+    def get_max_ids(self):
+        """Get the maximum id in the DAGMC model for each entity type.
+
+        returns:
+            max_ids (dict): {entity dimension (int): max_id (int)}
+        """
+        # dim 0: vertex
+        # dim 1: edge
+        # dim 2: surface
+        # dim 3: volume
+        # dim 4: group
+        max_ids = {dim: 0 for dim in range(5)}
+        root_set = self.mb.get_root_set()
+        geom_sets = self.mb.get_entities_by_type_and_tag(
+            root_set, types.MBENTITYSET, self.geom_dim_tag, [None]
+        )
+
+        for entity_set in geom_sets:
+            dim = self.mb.tag_get_data(self.geom_dim_tag, entity_set)[0][0]
+            gid = self.mb.tag_get_data(self.global_id_tag, entity_set)[0][0]
+            max_ids[dim] = max(max_ids[dim], gid)
+
+        return max_ids
+
+    def reassign_global_ids(self, offsets):
+        """Renumber entities, starting from the offset value for that
+            dimension.
+
+        Arguments:
+            offsets (dict): {entity dimension (int): id_offset (int)}
+        """
+        root_set = self.mb.get_root_set()
+        geom_sets = self.mb.get_entities_by_type_and_tag(
+            root_set, types.MBENTITYSET, self.geom_dim_tag, [None]
+        )
+
+        for entity_set in geom_sets:
+            dim = self.mb.tag_get_data(self.geom_dim_tag, entity_set)[0][0]
+            current_gid = self.mb.tag_get_data(self.global_id_tag, entity_set)[
+                0
+            ][0]
+            self.mb.tag_set_data(
+                self.global_id_tag,
+                entity_set,
+                current_gid + offsets.get(dim, 0),
+            )
+
+    def write_file(self, filename):
+        """Save to file.
+
+        Arguments:
+            filename (str): Path to save the file to. All filetypes supported
+                by MOAB are available. Write to h5m for use with DAGMC.
+        """
+        self.mb.write_file(filename)
+
+
+def merge_dagmc_files(models_to_merge):
```

Comment:
Doesn't seem like the I/O stream here actually uses any files
```suggestion
def merge_dagmc_models(models_to_merge):
```